api/src/main/java/org/pmiops/workbench/auth/ServiceAccounts.java

```diff
@@ -13,14 +17,46 @@
   private static final String SIGN_JWT_URL_FORMAT =
       "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/%s:signJwt";
 
+  /**
+   * Returns an appropriate set of service credentials based on the current operating environment.
+   *
+   * <p>When running a local server, Application Default Credentials (ADCs) are used.
+   *
+   * <p>When running in App Engine, we use AppIdentityService to return an instance of
+   * AppEngineCredentials. Unfortunately, if we use GoogleCredentials.getApplicationDefault() from
+   * within an App Engine environment, the returned credentials will be an instance of
+   * ComputeEngineCredentials, which doesn't support scoped access tokens. Frustratingly, the call
+   * to .createScoped will silently proceed by doing nothing -- meaning we only learn about the
+   * error once an attempt to use these credentials fails in a downstream service due to bad scopes.
+   *
+   * <p>See https://github.com/googleapis/google-auth-library-java/issues/272 and
+   * https://github.com/googleapis/google-auth-library-java/issues/172 for reference; this seems to
+   * be a common pain point for users of the com.google.auth.oauth2 library.
+   *
+   * @param scopes
+   * @return
+   * @throws IOException
+   */
+  private static GoogleCredentials getScopedServiceCredentials(List<String> scopes)
+      throws IOException {
+    if (SystemProperty.environment.value().equals(SystemProperty.Environment.Value.Development)) {
```

Comment:
Could use the just-removed `WorkbenchEnvironment` here if it still existed.  Would it make sense to bring it back?

Reply:
Yeah, I thought about that... I'm not convinced that a separate dependency-injected class is worth it for the single usage instance here. If there's no other obvious advantage that I'm missing, I'm inclined to keep the "am I in dev" check local here.

[Update] as a last-ditch effort, I tried seeing if the AppIdentityService branch would work correctly even when running locally. I tested this out locally and it does work fine (I suspect that somewhere under the hood there is a fallback call to GoogleCredentials.getApplicationDefault), but I'm not confident enough to push out that change. I'd rather leave as-is for now and possibly follow-up later with that simplification.